### PR TITLE
[ADP-3366] Refactor genesis files creation in local cluster

### PR DESCRIPTION
**Pull request by paolino**
- [x] refactor GenesisFiles creation
- [x] move the GenesisFiles module inside the Node namespace

ADP-3366

### CODE REVIEW

File: lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenesisFiles.hs [hunk 1]
```diff
@@ -222,49 +318,31 @@ generateGenesis initialFunds genesisMods = do
...
                           -- (and probably for the BFT node pre-babbage):
                           sgGenDelegs =
-                            fromRight (error "invalid sgGenDelegs") . Aeson.eitherDecode
-                                $ Aeson.encode
-                                    [aesonQQ|
-                    {"91612ee7b158dc64871a959060973d0f2b8fb6e85ae960f03b8640ac": {
-                        "delegate": "180b3fae61789f61cbdbc69e5f8e1beae9093aa2215e482dc8d89ec9",
-                        "vrf": "e9ef3b5d81d400eb046de696354ff8e84122f505e706e3c86a361cce919a686e"
-                    }}|]
+                            fromRight (error "invalid sgGenDelegs")
```
Comment (Anviking) [inline lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenesisFiles.hs]: > We need this to submit MIR certs (and probably for the BFT node pre-babbage):

🤔 Maybe it's possible to leave `sgGenDelegs` empty now then?

Comment (paolino) [reply lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenesisFiles.hs]: Judging from the comment yes. Trying...

File: lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenesisFiles.hs [hunk 2]
```diff
@@ -100,26 +121,88 @@ import Data.Time.Clock
...
+
+instance FFunctor GenesisRecord where ffmap = ffmapDefault
+instance FFoldable GenesisRecord where ffoldMap = ffoldMapDefault
+instance FTraversable GenesisRecord where ftraverse = gftraverse
+
+instance FZip GenesisRecord where fzipWith = gfzipWith
+
+deriving stock instance Show (GenesisRecord FileOf)
+
+-- type GenesisFiles = GenesisRecord FileOf
```
Comment (Anviking) [inline lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenesisFiles.hs]:
```suggestion
```

File: lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenNodeConfig.hs [hunk 3]
```diff
@@ -43,9 +43,11 @@ import Cardano.Wallet.Launch.Cluster.FileOf
     , absFilePathOf
     , toFilePath
     )
-import Cardano.Wallet.Launch.Cluster.GenesisFiles
-    ( GenesisFiles (..)
+import Cardano.Wallet.Launch.Cluster.Node.GenesisFiles
+    ( GenesisFiles
+    , GenesisRecord (..)
     )
+
```
Comment (Anviking) [inline lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenNodeConfig.hs]:
```suggestion
```

File: lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenesisFiles.hs [hunk 4]
```diff
@@ -221,50 +304,19 @@ generateGenesis initialFunds genesisMods = do
...
-                , conwayGenesis = FileOf conwayGenesis
+        produceGenesis cfgClusterConfigs cfgClusterDir
+            $ GenesisRecord
+                { byronGenesis =
+                    Const
+                        $ key "startTime"
+                            .~ toJSON
+                                (round @_ @Int $ utcTimeToPOSIXSeconds systemStart)
+                , shelleyGenesis = Const
+                    $ \_ -> toJSON shelleyGenesisData
```
Comment (Anviking) [inline lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenesisFiles.hs:317]: So we could remove the shelley template? Or are you envisioning users being able to supply their own templates and the cluster startup doing only the minimal modifications?

**Review by Anviking — APPROVED** [review]

File: lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenesisFiles.hs (hunk 4 above)
Comment (paolino) [reply lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenesisFiles.hs:317]: That would be my idea yes

**Pull request merged**